test/pkg/environment/environment.go

```diff
@@ -40,15 +41,37 @@ import (
 	"github.com/aws/karpenter/pkg/utils/project"
 )
 
+type AWSEnvironment struct {
```

Comment:
Can we make this a new file? Sounds like we may want to port part of this to karpenter-core?